app/models/active_lead_provider.rb

```diff
@@ -10,12 +10,12 @@ class ActiveLeadProvider < ApplicationRecord
   validates :contract_period_year,
             presence: { message: 'Choose a contract period' },
             uniqueness: { scope: :lead_provider_id, message: 'Contract period and lead provider must be unique' }
-
   validates :lead_provider_id, presence: { message: 'Choose a lead provider' }
 
   scope :for_contract_period, ->(year) { where(contract_period_year: year) }
```

Comment:
I would be keen to replace all references to `for_contract_period` with the scope `for_contract_period_year` because I think it's clearer (I started it but it's too unwieldy for this PR). SchoolPartnership also has a `for_contract_period` scope which could be updated too.